packages/discord.js/src/structures/Presence.js

```diff
@@ -254,7 +254,8 @@ class Activity {
         this.type === activity.type &&
         this.url === activity.url &&
         this.state === activity.state &&
-        this.details === activity.details)
+        this.details === activity.details &&
+        (this.emoji?.id ? this.emoji.id === activity.emoji?.id : this.emoji?.name === activity.emoji?.name))
```

Comment:
```suggestion
        this.emoji?.id === activity.emoji?.id &&
        this.emoji?.name === activity.emoji?.name)
```

Reply:
Won't this falsely update if the emoji is a guild emoji and its name is updated?

Reply:
How is that a false update?

Reply:
I'm kinda wavering back and forth, but you would know better up to you

Reply:
I would say to change it to this. By the way, modifying the name of a custom emoji will not fire a `PRESENCE_UPDATE` event.

Reply:
but updating presence yes
and some old presence can have old name